src/lib.rs

```diff
@@ -887,16 +908,28 @@ impl InnerConnection {
             return Ok(());
         }
         self.remove_hooks();
+        let mut shared_handle = self.interrupt_lock.lock().unwrap();
```

Comment:
Maybe we should close the connection even is lock is poisoned ?

Reply:
I think this could lead to double sqlite3_close calls, which is probably worse than panicking (it's a memory safety problem).

Reply:
I don't see how InnerConnection.close can be called twice ?

Reply:
The only two places that could poison the lock are `InnerConnection::close()` or `InterruptHandle::interrupt()`. ISTM like the former is more likely since it does a lot more. At the moment, there should be no way for it to happen twice, but it seems more likely to me than `interrupt()`.